feature/dynamodb/enhancedclient/schema.go

```diff
@@ -0,0 +1,128 @@
+package enhancedclient
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type Schema[T any] struct {
+	options      SchemaOptions
+	cachedFields *CachedFields
+	enc          *Encoder[T]
+	dec          *Decoder[T]
+	typ          reflect.Type
+
+	//generators map[string]enhancedclient2.Generator[T]
+	extensions map[ExecutionPhase][]Extension
+
+	// common
```

Comment:
if new attributes are added to DynamoDB model, would we also need tho add them here?